frontend/packages/dev-console/src/components/health-checks/AddHealthChecks.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form, Button } from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import {
+  ContainerDropdown,
+  history,
+  PageHeading,
+  ResourceLink,
+  ResourceIcon,
+} from '@console/internal/components/utils';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FormFooter } from '@console/shared';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import HealthChecks from './HealthChecks';
+import Helmet from 'react-helmet';
+import { ContainerModel } from '@console/internal/models';
+import './AddHealthChecks.scss';
+import { getHealthChecksData } from './create-health-checks-probe-utils';
+
+type AddHealthChecksProps = {
+  resource?: K8sResourceKind;
+  currentContainer: string;
+};
+
+const AddHealthChecks: React.FC<FormikProps<FormikValues> & AddHealthChecksProps> = ({
+  resource,
+  currentContainer,
+  handleSubmit,
+  handleReset,
+  dirty,
+  errors,
+  status,
+  isSubmitting,
+  setFieldValue,
+}) => {
+  const [currentKey, setCurrentKey] = React.useState(currentContainer);
+  const containers = resource?.spec?.template?.spec?.containers;
+  const healthCheckAdded = _.every(
+    containers,
+    (container) => container.readinessProbe || container.livenessProbe || container.startupProbe,
+  );
+  const containersByKey = _.keyBy(containers, 'name');
+  const pageTitle = healthCheckAdded ? 'Edit Health Checks' : 'Add Health Checks';
+
+  const handleSelectContainer = (containerName: string) => {
+    const containerIndex = _.findIndex(resource.spec.template.spec.containers, [
+      'name',
+      containerName,
+    ]);
+    setCurrentKey(containerName);
+    setFieldValue('containerName', containerName);
+    setFieldValue('healthChecks', getHealthChecksData(resource, containerIndex));
+    history.push(
+      `/k8s/ns/${resource.metadata.namespace}/${resource.kind}/${resource.metadata.name}/containers/${containerName}/health-checks`,
+    );
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle}</title>
+      </Helmet>
+      <PageHeading
+        title={
+          <>
+            {pageTitle}
+            <Button
+              variant="link"
+              component="a"
+              href="https://docs.openshift.com/container-platform/3.11/dev_guide/application_health.html"
```

Comment:
I really don't think we want to be pointing to old documentation. Not sure what UX had in mind. Make an issue to get an updated link before release.

Reply:
When I made the first comment about matching UX design, I wasn't sure if they provided you with a proper link. If we don't have one, we can also omit this link for now until we know what to put. Either way we need an issue to track updating this URL.

Reply:
Created an issue to update the doc link when it is available for 4.5. https://issues.redhat.com/browse/ODC-3620